components/profile.js

```diff
@@ -65,57 +84,401 @@ SteamCommunity.prototype.editProfile = function(settings, callback) {
 
 			switch(i) {
 				case 'name':
-					values.personaName = settings[i];
+					out[8].value = settings[i];
 					break;
 
 				case 'realName':
-					values.real_name = settings[i];
+					out[9].value = settings[i];
 					break;
 
 				case 'summary':
-					values.summary = settings[i];
+					out[14].value = settings[i];
 					break;
 
 				case 'country':
-					values.country = settings[i];
+					out[10].value = settings[i];
 					break;
 
 				case 'state':
-					values.state = settings[i];
+					out[11].value = settings[i];
 					break;
 
 				case 'city':
-					values.city = settings[i];
+					out[12].value = settings[i];
 					break;
 
 				case 'customURL':
-					values.customURL = settings[i];
+					out[13].value = settings[i];
 					break;
 
 				case 'background':
 					// The assetid of our desired profile background
-					values.profile_background = settings[i];
+					out[15].value = settings[i];
 					break;
 
 				case 'featuredBadge':
 					// Currently, game badges aren't supported
-					values.favorite_badge_badgeid = settings[i];
+						out[16].value = settings[i];
 					break;
 
 				case 'primaryGroup':
-					if(typeof settings[i] === 'object' && settings[i].getSteamID64) {
-						values.primary_group_steamid = settings[i].getSteamID64();
+					if (typeof settings[i] === 'object' && settings[i].getSteamID64) {
+						out[18].value = settings[i].getSteamID64();
 					} else {
-						values.primary_group_steamid = new SteamID(settings[i]).getSteamID64();
+						out[18].value = new SteamID(settings[i]).getSteamID64();
 					}
 
 					break;
 
-				// TODO: profile showcases
+				case 'showcases':
+					for (var type in settings[i]) {
+						//Variable used to easily make request to`ajaxsetshowcaseconfig` for showcases like trade, items, ...
+						var showcaseconfig = {
+							"supplied": false,
+							"numberofrequests": 0,
+							"showcasetype": 0,
+							"itemarray": []
+						};
+
+						switch (settings[i][type].showcase) {
+
+							case 'infobox':
+								out.push({
+									"name": "profile_showcase[]",
+									"value": "8"
+								});
+								out.push({
+									"name": "rgShowcaseConfig[8][0][title]",
+									"value": settings[i][type]["values"]["title"]
```

Comment:
If you have already decided to access objects properties with `obj.property` syntax, you should stick to it.